cmd/argocd/commands/logout.go

```diff
@@ -36,6 +45,54 @@ $ argocd logout
 				log.Fatalf("Nothing to logout from")
 			}
 
+			token := localCfg.GetToken(context)
+			if token == "" {
+				log.Fatalf("Error in getting token from context")
+			}
+
+			client := &http.Client{}
+
+			tlsTestResult, err := grpc_util.TestTLS(context, DialTime)
+			errors.CheckError(err)
+			if !tlsTestResult.TLS {
+				if !globalClientOpts.PlainText {
+					if !cli.AskToProceed("WARNING: server is not configured with TLS. Proceed (y/n)? ") {
+						os.Exit(1)
+					}
+					globalClientOpts.PlainText = true
+				}
+			} else if tlsTestResult.InsecureErr != nil {
+				if !globalClientOpts.Insecure {
+					if !cli.AskToProceed(fmt.Sprintf("WARNING: server certificate had error: %s. Proceed insecurely (y/n)? ", tlsTestResult.InsecureErr)) {
+						os.Exit(1)
+					}
+					globalClientOpts.Insecure = true
+				}
+			}
+
+			scheme := "https"
+			if globalClientOpts.PlainText {
+				scheme = strings.TrimSuffix(scheme, "s")
+			} else if globalClientOpts.Insecure {
+				client.Transport = &http.Transport{
+					TLSClientConfig: &tls.Config{
+						InsecureSkipVerify: true,
+					},
+				}
+			}
+
+			logoutURL := fmt.Sprintf("%s://%s%s", scheme, context, common.LogoutEndpoint)
+			req, err := http.NewRequest(http.MethodPost, logoutURL, nil)
+			errors.CheckError(err)
+			cookie := &http.Cookie{
+				Name:  common.AuthCookieName,
+				Value: token,
+			}
+			req.AddCookie(cookie)
+
+			_, err = client.Do(req)
```

Comment:
This http call needs to be made after reconfirming from the user, needs to be moved after checking the`canLogout`.